developer-docs-site/static/messageboard_demo/client/README.md

```diff
@@ -0,0 +1,187 @@
+
+```tsx
+public(script) fun send_pinned_message(
+        account: &signer, board_addr: address, message: vector<u8>
+    ) acquires MessageChangeCapability, MessageChangeEventHandle, CapBasedMB {
+        let cap = borrow_global<MessageChangeCapability>(Signer::address_of(account));
+        assert!(cap.board == board_addr, EACCOUNT_NO_NOTICE_CAP);
+        let board = borrow_global_mut<CapBasedMB>(board_addr);
+        board.pinned_post = message;
+       ...
+    }
+```
+
+To provide participant the capability resource, the board creator creates the MessageChangeCapability and offer the capability to a participant’s address. 
+
+```tsx
+public(script) fun add_participant(account: &signer, participant: address) acquires MessageCapEventHandle {
+    let board_addr = Signer::address_of(account);
+    Offer::create(account, MessageChangeCapability{ board: board_addr }, participant);
+...
+}
+```
+
+The participant can then claim this capability and then move it to their own account. Later on, the participant can use this resource stored in their account to modify the pinned_post
+
+```tsx
+let notice_cap = Offer::redeem<MessageChangeCapability>(
+            account, board);
+move_to(account, notice_cap);
+```
+
+Move also support emitting events and these events will be stored in event stream that can be queried through API request. For example, any one can send a message to the messageboard if they don’t want to modify the pinned post. The contract will emit an event and add the event to the board’s event stream. Thus anyone subscribe to the event stream of the board can see all the messages. 
+
+```tsx
+public(script) fun send_message_to(
+    account: signer, board_addr: address, message: vector<u8>
+) acquires MessageChangeEventHandle {
+    let event_handle = borrow_global_mut<MessageChangeEventHandle>(board_addr);
+    Event::emit_event<MessageChangeEvent>(
+        &mut event_handle.change_events,
+        MessageChangeEvent{
+            message,
+            participant: Signer::address_of(&account)
+        }
+    );
+}
+```
+
+### ACL-Based Messageboard
+
+[ACL-Based messageboard](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/messageboard/sources/ACLMessageBoard.move) is similar to the capability based messageboard. The main difference is it uses ACL to control the access, which is essentially a list of the addresses. 
+
+To add a new participant, the board creator adds the participant into the ACL list under his account. 
+
+```tsx
+public(script) fun add_participant(account: &signer, participant: address) acquires ACLBasedMB {
+    let board = borrow_global_mut<ACLBasedMB>(Signer::address_of(account));
+    ACL::add(&mut board.participants, participant);
+}
+```
+
+When a participant want to change the pinned message, the participant first need to borrow the board resource under creator’s account and check if the participant is in the ACL. Then it borrows mutable board to modify the pinned_post. 
```

Comment:
When a participant wants to change the pinned message, the participant must first borrow the board resource under creator’s account, and check if the participant is in the ACL. Then the participant borrows mutable board to modify the pinned_post.

Reply:
updated